barretenberg/cpp/src/barretenberg/crypto/merkle_tree/indexed_tree/content_addressed_indexed_tree.hpp

```diff
@@ -1144,7 +1195,7 @@ void ContentAddressedIndexedTree<Store, HashingPolicy>::update_leaf_and_hash_to_
     // 3. Write the new node value
     index_t index = leaf_index;
     uint32_t level = depth_;
-    fr new_hash = HashingPolicy::hash(leaf.get_hash_inputs());
+    fr new_hash = leaf.value.is_empty() ? 0 : HashingPolicy::hash(leaf.get_hash_inputs());
```

Comment:
Should we use fr::zero() instead of 0?

Reply:
I think recently there was a group of PRs to change fr::zero() to 0. The rationale is that the constructor of fr is constexpr, so the cost is the same (compile-time), while readability is (subjectively) worse. I have no preference personally.

Reply:
Me neither, but since in other places of this file fr::zero is used, I don't mind making it the same